frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/network-attachment-definition.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as _ from 'lodash';
+import { sortable } from '@patternfly/react-table';
+import { ListPage, Table, TableData, TableRow } from '@console/internal/components/factory';
+import { Kebab, ResourceKebab, ResourceLink } from '@console/internal/components/utils';
+import { NamespaceModel } from '@console/internal/models';
+import { K8sResourceKindReference } from '@console/internal/module/k8s';
+import { NetworkAttachmentDefinitionModel } from '@console/shared/src/models';
+import { NetworkAttachmentDefinitionKind } from '@console/shared/src/types';
+
+const { common } = Kebab.factory;
+const menuActions = [...common];
+
+export const NetworkAttachmentDefinitionReference: K8sResourceKindReference =
+  'NetworkAttachmentDefinition';
+
+const tableColumnClasses = [
+  classNames('col-lg-4', 'col-md-4', 'col-sm-6', 'col-xs-6'),
+  classNames('col-lg-4', 'col-md-4', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-4', 'col-md-4', 'col-sm-6', 'col-xs-6'),
+  Kebab.columnClass,
+];
+
+const NetworkAttachmentDefinitionsHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Namespace',
+      sortField: 'metadata.namespace',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Type',
+      sortFunc: 'string',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: '',
+      props: { className: tableColumnClasses[3] },
+    },
+  ];
+};
+NetworkAttachmentDefinitionsHeader.displayName = 'NADTableHeader';
+
+const NetworkAttachmentDefinitionsRow: React.FC<NetworkAttachmentDefinitionsRowProps> = ({
+  obj,
+  index,
+  key,
+  style,
+  namespace,
+}) => {
+  const config = JSON.parse(_.get(obj, 'spec.config'));
```

Comment:
can you put this somewhere to utils and also use `NetworkAttachmentDefinitionConfig` as return value type ?

Reply:
Done!